dist/stampit.js

```diff
@@ -3829,24 +3835,29 @@ function compose(factories) {
  * prototypes that are passed in or composed.
  *
  * @param  {Object} [methods] A map of method names and bodies for delegation.
- * @param  {Object} [state]   A map of property names and values to clone for each new object.
+ * @param  {Object} [refs]   A map of property names and values to be mixed into each new object.
  * @param  {Function} [enclose] A closure (function) used to create private data and privileged methods.
+ * @param  {Object} [props]   An object to be deeply cloned into each newly stamped object.
  * @return {Function} factory A factory to produce objects using the given prototypes.
  * @return {Function} factory.create Just like calling the factory function.
  * @return {Object} factory.fixed An object map containing the fixed prototypes.
- * @return {Function} factory.methods Add methods to the methods prototype. Chainable.
- * @return {Function} factory.state Add properties to the state prototype. Chainable.
- * @return {Function} factory.enclose Add or replace the closure prototype. Not chainable.
- */
-stampit = function stampit(methods, state, enclose) {
-  var fixed = { methods: {}, state: {}, enclose: [] };
+ * @return {Function} factory.methods Add methods to the prototype. Chainable.
+ * @return {Function} factory.refs Add references to the prototype. Chainable.
+ * @return {Function} factory.state Alias to refs().
+ * @return {Function} factory.enclose Add a closure which called on object instantiation. Chainable.
+ * @return {Function} factory.props Add deeply cloned properties to the produced objects. Chainable.
+ */
+stampit = function stampit(methods, refs, enclose, props) {
+  var fixed = {methods: {}, refs: {}, enclose: [], props: {}};
+  fixed.state = fixed.refs; // Backward compatibility.
   addMethods(fixed, methods);
-  addState(fixed, state);
+  addRefs(fixed, refs);
   addEnclose(fixed, enclose);
+  addProps(fixed, props);
```

Comment:
This needs to go before the `enclose()` methods so that closures can access stamp props.